From 2973fef5f19231650bb38491e631a23454e2376d Mon Sep 17 00:00:00 2001
From: Daiki Ueno <ueno@gnu.org>
Date: Thu, 26 Feb 2026 08:03:38 +0900
Subject: [PATCH 02/11] BACKPORT: nettle: include <nettle/sha[12].h> instead of
 deprecated <nettle/sha.h>

Signed-off-by: Daiki Ueno <ueno@gnu.org>
(cherry picked from commit ac683ce5149108dde63954cf149dd4a48f410f6f)
Signed-off-by: Xi Ruoyao <xry111@xry111.site>
---
 lib/accelerated/aarch64/hmac-sha-aarch64.c | 3 ++-
 lib/accelerated/aarch64/sha-aarch64.c      | 3 ++-
 lib/accelerated/aarch64/sha-aarch64.h      | 3 ++-
 lib/accelerated/x86/hmac-padlock.c         | 3 ++-
 lib/accelerated/x86/hmac-x86-ssse3.c       | 3 ++-
 lib/accelerated/x86/sha-padlock.c          | 3 ++-
 lib/accelerated/x86/sha-padlock.h          | 3 ++-
 lib/accelerated/x86/sha-x86-ssse3.c        | 3 ++-
 lib/accelerated/x86/sha-x86.h              | 3 ++-
 lib/nettle/mac.c                           | 3 ++-
 10 files changed, 20 insertions(+), 10 deletions(-)

diff --git a/lib/accelerated/aarch64/hmac-sha-aarch64.c b/lib/accelerated/aarch64/hmac-sha-aarch64.c
index 46bf1ca3c..6261180f0 100644
--- a/lib/accelerated/aarch64/hmac-sha-aarch64.c
+++ b/lib/accelerated/aarch64/hmac-sha-aarch64.c
@@ -28,7 +28,8 @@
 #include "gnutls_int.h"
 #include "hash_int.h"
 #include "errors.h"
-#include <nettle/sha.h>
+#include <nettle/sha1.h>
+#include <nettle/sha2.h>
 #include <nettle/hmac.h>
 #include <nettle/macros.h>
 #include "sha-aarch64.h"
diff --git a/lib/accelerated/aarch64/sha-aarch64.c b/lib/accelerated/aarch64/sha-aarch64.c
index c46e519d9..42adf283c 100644
--- a/lib/accelerated/aarch64/sha-aarch64.c
+++ b/lib/accelerated/aarch64/sha-aarch64.c
@@ -25,7 +25,8 @@
 #include "gnutls_int.h"
 #include <gnutls/crypto.h>
 #include "errors.h"
-#include <nettle/sha.h>
+#include <nettle/sha1.h>
+#include <nettle/sha2.h>
 #include <nettle/macros.h>
 #include <nettle/nettle-meta.h>
 #include "sha-aarch64.h"
diff --git a/lib/accelerated/aarch64/sha-aarch64.h b/lib/accelerated/aarch64/sha-aarch64.h
index 25eb4f3d6..c5c227a72 100644
--- a/lib/accelerated/aarch64/sha-aarch64.h
+++ b/lib/accelerated/aarch64/sha-aarch64.h
@@ -1,7 +1,8 @@
 #ifndef GNUTLS_LIB_ACCELERATED_AARCH64_SHA_AARCH64_H
 #define GNUTLS_LIB_ACCELERATED_AARCH64_SHA_AARCH64_H
 
-#include <nettle/sha.h>
+#include <nettle/sha1.h>
+#include <nettle/sha2.h>
 
 extern const struct nettle_hash aarch64_sha1;
 extern const struct nettle_hash aarch64_sha224;
diff --git a/lib/accelerated/x86/hmac-padlock.c b/lib/accelerated/x86/hmac-padlock.c
index 2c068372d..ac91d20ec 100644
--- a/lib/accelerated/x86/hmac-padlock.c
+++ b/lib/accelerated/x86/hmac-padlock.c
@@ -27,7 +27,8 @@
 #include "gnutls_int.h"
 #include "hash_int.h"
 #include "errors.h"
-#include <nettle/sha.h>
+#include <nettle/sha1.h>
+#include <nettle/sha2.h>
 #include <nettle/hmac.h>
 #include <nettle/macros.h>
 #include <nettle/memxor.h>
diff --git a/lib/accelerated/x86/hmac-x86-ssse3.c b/lib/accelerated/x86/hmac-x86-ssse3.c
index 9b1955249..dec9c2688 100644
--- a/lib/accelerated/x86/hmac-x86-ssse3.c
+++ b/lib/accelerated/x86/hmac-x86-ssse3.c
@@ -27,7 +27,8 @@
 #include "gnutls_int.h"
 #include "hash_int.h"
 #include "errors.h"
-#include <nettle/sha.h>
+#include <nettle/sha1.h>
+#include <nettle/sha2.h>
 #include <nettle/hmac.h>
 #include <nettle/macros.h>
 #include "aes-x86.h"
diff --git a/lib/accelerated/x86/sha-padlock.c b/lib/accelerated/x86/sha-padlock.c
index 2a4a4de45..b2c4f095e 100644
--- a/lib/accelerated/x86/sha-padlock.c
+++ b/lib/accelerated/x86/sha-padlock.c
@@ -24,7 +24,8 @@
 #include "gnutls_int.h"
 #include "hash_int.h"
 #include "errors.h"
-#include <nettle/sha.h>
+#include <nettle/sha1.h>
+#include <nettle/sha2.h>
 #include <nettle/hmac.h>
 #include <nettle/macros.h>
 #include "aes-padlock.h"
diff --git a/lib/accelerated/x86/sha-padlock.h b/lib/accelerated/x86/sha-padlock.h
index 990c02d98..5247959be 100644
--- a/lib/accelerated/x86/sha-padlock.h
+++ b/lib/accelerated/x86/sha-padlock.h
@@ -1,7 +1,8 @@
 #ifndef GNUTLS_LIB_ACCELERATED_X86_SHA_PADLOCK_H
 #define GNUTLS_LIB_ACCELERATED_X86_SHA_PADLOCK_H
 
-#include <nettle/sha.h>
+#include <nettle/sha1.h>
+#include <nettle/sha2.h>
 
 void padlock_sha1_oneshot(void *ctx, const void *inp, size_t len);
 void padlock_sha256_oneshot(void *ctx, const void *inp, size_t len);
diff --git a/lib/accelerated/x86/sha-x86-ssse3.c b/lib/accelerated/x86/sha-x86-ssse3.c
index 28d7930e6..c0dbce62e 100644
--- a/lib/accelerated/x86/sha-x86-ssse3.c
+++ b/lib/accelerated/x86/sha-x86-ssse3.c
@@ -25,7 +25,8 @@
 #include <gnutls/crypto.h>
 #include "errors.h"
 #include "aes-x86.h"
-#include <nettle/sha.h>
+#include <nettle/sha1.h>
+#include <nettle/sha2.h>
 #include <nettle/macros.h>
 #include <nettle/nettle-meta.h>
 #include "sha-x86.h"
diff --git a/lib/accelerated/x86/sha-x86.h b/lib/accelerated/x86/sha-x86.h
index d96936663..a01e0e639 100644
--- a/lib/accelerated/x86/sha-x86.h
+++ b/lib/accelerated/x86/sha-x86.h
@@ -1,7 +1,8 @@
 #ifndef GNUTLS_LIB_ACCELERATED_X86_SHA_X86_H
 #define GNUTLS_LIB_ACCELERATED_X86_SHA_X86_H
 
-#include <nettle/sha.h>
+#include <nettle/sha1.h>
+#include <nettle/sha2.h>
 
 extern const struct nettle_hash x86_sha1;
 extern const struct nettle_hash x86_sha224;
diff --git a/lib/nettle/mac.c b/lib/nettle/mac.c
index b6be1c913..2c76fb56a 100644
--- a/lib/nettle/mac.c
+++ b/lib/nettle/mac.c
@@ -29,7 +29,8 @@
 #include <nettle/md5.h>
 #include <nettle/md2.h>
 #include <nettle/ripemd160.h>
-#include <nettle/sha.h>
+#include <nettle/sha1.h>
+#include <nettle/sha2.h>
 #include <nettle/sha3.h>
 #ifndef HAVE_NETTLE_SHA3_128_SHAKE_OUTPUT
 #include "int/sha3-shake.h"
-- 
2.53.0

